### PR TITLE
Move async queue length metrics to dequeuer

### LINKED ISSUE #2280: Move queue metrics from operator to dequeuer

### PR DESCRIPTION
**Pull request by miguelvr**
closes #2280 

---

checklist:

- [x] run `make test` and `make lint`
- [ ] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

**Review by RobertLucian — APPROVED** [review]
LGTM!

File: cmd/dequeuer/main.go [hunk 1]
```diff
@@ -180,7 +180,10 @@ func main() {
 			TargetURL:  targetURL,
 		}
 
-		asyncStatsReporter := dequeuer.NewAsyncPrometheusStatsReporter()
+		asyncStatsReporter := dequeuer.NewAsyncPrometheusStatsReporter(awsClient.SQS(), queueURL, log)
+		stopStatsReporter := asyncStatsReporter.Start()
```
Comment (deliahu) [inline cmd/dequeuer/main.go:184]: Will this run on every async replica?

Comment (miguelvr) [reply cmd/dequeuer/main.go:184]: that's a very good point, maybe it's not worth it to move this metric out of the operator?

Comment (deliahu) [reply cmd/dequeuer/main.go:184]: We can discuss this next week

**Comment by deliahu** [comment]
Closing this, as we discussed

**Pull request closed without merging**